src/Extend/ApiSerializer.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Api\Serializer\AbstractSerializer;
+use Flarum\Extension\Extension;
+use Flarum\Foundation\ContainerUtil;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Container\Container;
+
+class ApiSerializer implements ExtenderInterface
+{
+    private $serializerClass;
+    private $attributeHandlers = [];
+    private $settings = [];
+    private $relationships = [];
+
+    /**
+     * @param string $serializerClass The ::class attribute of the serializer you are modifying.
+     *                                This serializer should extend from \Flarum\Api\Serializer\AbstractSerializer.
+     */
+    public function __construct(string $serializerClass)
+    {
+        $this->serializerClass = $serializerClass;
+    }
+
+    /**
+     * Add to or modify the attributes array of this serializer.
+     *
+     * @param callable|string $callback
+     *
+     * The callback can be a closure or an invokable class, and should accept:
+     * - $attributes: An array of existing attributes.
+     * - $model: An instance of the model being serialized.
+     * - $serializer: An instance of this serializer.
+     *
+     * The callable should return:
+     * - An array of additional attributes to merge with the existing array.
+     *   Or a modified $attributes array.
+     *
+     * @return self
+     */
+    public function attributes($callback)
```

Comment:
it would have to be `attribute($nameOrCallback, $value = null)` because if you are going to use a callback, there is no point in providing an attribute name.

Or we could provide two methods, but I'm not sure about the naming

Reply:
The callback would return the value assigned to the attribute. I'm thinking we do this kind of like the Model extender.

Reply:
but then wouldn't that mean, that if one chose to use an invokable class, and wanted to add multiple attributes they would have to create multiple classes instead of one ?

Reply:
True... I've been thinking that we might want to support laravel's [class, methodName] for callbacks.  @flarum/core any preference?

Reply:
Idea @SychO9:

- We have an `attribute` method that takes an attribute name and a callback (that takes the serializer instance and model instance), and assigns an attribute with the value returned by that callback. Alternatively, 
- We have a `mutate` method that takes the serializer instance and applies a random callback to it
- We implement `attribute` by assembling all the `attribute` calls within an extender into one mutator

Reply:
is this what you mean ?
```php
(new Extend\ApiSerializer())
    ->attribute('attrName', function ($attributes, $serializer, $model) {
        return 'a_value';
    })->mutate(function ($attributes, $serializer, $model) {
        $attributes['attr'] = 'a_value';
        $attributes['attr2'] = 'a_value2';

        // possibly some other logic..

        return $attributes;
    })
```

Reply:
Yeah something like that!